### PR TITLE
Added functions for data summary

### PR DESCRIPTION
**Pull request by ricalanis**
Modified three files:

main.js:
self.valueCount -  function(field): returns an array of the form [ {name: unique ids in field, y: count of the given unique id)} ]
self.valueMoney - returns an array of the form [ {name: unique ids in the given field, y: sum of "cantidad" given the unique id)}]

api.js:
countByFieldData - tool function to generate the sum of the given fields. returns an array of the form [{unique ids:count}]
moneyByFieldData - tool function to create the array in main.js

utilities.js:
convertHashtoID - converts the hash created by countbyFieldData in api.js to the structure needed by main.js

### CODE REVIEW

**Comment by ricalanis** [comment]
@jimenezjrs ¿Qué opinas? Te parece bien esta forma de traerme los valores?

File: app/scripts/controllers/main.js [hunk 1]
```diff
@@ -54,11 +54,19 @@ angular.module('financiamientoClimaticoApp')
       this.kpis.investment = 0;
     };
 
+    self.valueCount = function(field) {
+      return Utilities.convertHashtoID(Api.countByFieldData(this.results, field));
+    };
+
+    self.valueMoney = function(field) {
+      return Api.moneyByFieldData(this.results,field);
+    }
```
Comment (jimenezjrs) [inline app/scripts/controllers/main.js]: Falta ';'

File: app/scripts/services/api.js [hunk 2]
```diff
@@ -90,6 +90,18 @@ angular.module('financiamientoClimaticoApp')
...
+        return _.countBy(_.pluck(records, fieldName));
+      },
+      moneyByFieldData: function(records, fieldName){
+        var groups = _(records).groupBy(fieldName);
+        var out = _(groups).map(function(g, key) {
+          return { name: key, 
+                   y: _(g).reduce(function(m,x) { 
+                    return m + parseInt(x.cantidad); }, 0) };
+        });
+        return out
```
Comment (jimenezjrs) [inline app/scripts/services/api.js]: Falta el ';'

File: app/scripts/services/utilities.js [hunk 3]
```diff
@@ -18,6 +18,16 @@ angular.module('financiamientoClimaticoApp')
         var recordPropertyValue = recordPropertyValue.toLowerCase();
         return  filterValue === null ||
         (filterValue.toLowerCase().indexOf(recordPropertyValue) !== -1);
+      },
+      convertHashtoID: function(countByHash){
+        var valueList = []
```
Comment (jimenezjrs) [inline app/scripts/services/utilities.js]: ;

File: app/scripts/services/utilities.js [hunk 4]
```diff
@@ -18,6 +18,16 @@ angular.module('financiamientoClimaticoApp')
...
+        Object.keys(countByHash).forEach(function (key) {
```
Comment (jimenezjrs) [inline app/scripts/services/utilities.js]: Angular forEach (https://docs.angularjs.org/api/ng/function/angular.forEach)

**Pull request merged**